File: entry_point/lua_entry_point/entry_point.lua
-- comments
function entry_point_function (n)
  return n;
end


function main(_arg)
  _arg = _arg or arg

  if #_arg ~= 1 then
    io.stderr:write("Usage: " .. arg[0] .." <arg1>\n")
    os.exit(1)
  end
end

--function main()
--  for idx_i=1,10,1
--  do
--    print("entry_point_function(" .. idx_i .. ")=<" .. tostring(entry_point_function(idx_i)) .. ">")
--  end
--  os.exit(1)
--end

main()

File: entry_point/scala_entry_point/entry_point.scala
object entry_point {
  def entry_point_function( n:Int ) : Int = {
    return n
  }

  def main(args: Array[String]): Unit = {
    if( args.length != 1 ) {
      Console.err.println("Usage: entry_point <arg1>");
      sys.exit(1)
    }
    for( idx_i <- 1 to 10) {
      Console.println("entry_point_function(" + idx_i + ")=<" + entry_point_function(idx_i) + ">")
    }
  }
}
